sys/targets/targets.go

```diff
@@ -924,6 +916,17 @@ func (target *Target) lazyInit() {
 	}
 	target.CFlags = newCFlags
 	target.CxxFlags = append(target.CFlags, commonCxxFlags...)
+	// In ESA/390 mode, the CPU is able to address only 31bit of memory but
+	// arithmetic operations are still 32bit
+	// Fix cflags by replacing compiler's -m32 option with -m31
+	if runtime.GOARCH == S390x {
```

Comment:
I think this should be done only for target.OS == TestOS.
You probably don't do it on s390, but generally we can build Fuchsia/NetBSD/etc with their cross-compilers here.

Reply:
Also, should we do this before checkFlagSupported checks above? Otherwise we will invoke compiler with -m32 and conclude all flags are unsupported.